minicluster/src/main/java/alluxio/master/LocalAlluxioCluster.java

```diff
@@ -128,6 +130,21 @@ public void initConfiguration() throws IOException {
             mHostname, mWorkDirectory).entrySet()) {
       ServerConfiguration.set(entry.getKey(), entry.getValue());
     }
+    if (mJournalType == JournalType.UFS) {
+      ServerConfiguration.set(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.UFS);
+    } else {
+      ServerConfiguration.set(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.EMBEDDED);
+      ServerConfiguration.set(PropertyKey.MASTER_EMBEDDED_JOURNAL_ADDRESSES,
+          String.format("%s:%s", mHostname,
+              ServerConfiguration.get(PropertyKey.MASTER_EMBEDDED_JOURNAL_PORT)));
+      ServerConfiguration.set(PropertyKey.MASTER_RPC_ADDRESSES, String.format("%s:%s", mHostname,
+          ServerConfiguration.get(PropertyKey.MASTER_RPC_PORT)));
+      if (ServerConfiguration.getBytes(PropertyKey.MASTER_JOURNAL_LOG_SIZE_BYTES_MAX) < 65) {
+        ServerConfiguration.set(PropertyKey.MASTER_JOURNAL_LOG_SIZE_BYTES_MAX,
+            PropertyKey.MASTER_JOURNAL_LOG_SIZE_BYTES_MAX.getDefaultValue());
+      }
+      ServerConfiguration.set(PropertyKey.MASTER_EMBEDDED_JOURNAL_SHUTDOWN_TIMEOUT, "100ms");
+    }
     ServerConfiguration.set(PropertyKey.TEST_MODE, true);
     ServerConfiguration.set(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.UFS);
```

Comment:
looks like we're overriding the journal type here anyway

Reply:
Removed, thanks